src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.DecimalValueSetFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static partial class ValueSetFactory
+    {
+        private sealed class DecimalValueSetFactory : NumericValueSetFactory<decimal, DecimalTC>, IValueSetFactory<decimal>, IValueSetFactory
+        {
+            public static new readonly DecimalValueSetFactory Instance = new DecimalValueSetFactory();
```

Comment:
What prevents a caller from using `NumericValueSetFactory<decimal, DecimalTC>.Instance` directly?

Reply:
Both `NumericValueSetFactory` and `DecimalTC` are private.

---
In reply to: [401124032](https://github.com/dotnet/roslyn/pull/42886#discussion_r401124032) [](ancestors = 401124032)